app/src/main/java/com/mobileproto/hireddit/hireddit/reddit/RedditSearcher.java

```diff
@@ -0,0 +1,160 @@
+    private RequestQueue queue;
+
+    private IndicoCallback<IndicoResult> indicoCallback = new IndicoCallback<IndicoResult>() {
+        @Override
+        public void handle(IndicoResult result) throws IndicoException {
+            getCommentFromKeywords(result.getKeywords().keySet());
+        }
+    };
+
+    public RedditSearcher(CommentCallback myCommentCallback, String spokenString, Context context) {
+        this.myCommentCallback = myCommentCallback;
+        this.spokenString = spokenString;
+        this.context = context;
+        String indicoApiKey = getApi(context);
+        this.indico = Indico.init(context, indicoApiKey, null);
+        this.queue = Volley.newRequestQueue(context);
+    }
+
+    public static String getApi(Context context) {
+        try {
+            AssetManager assetManager = context.getAssets();
+            InputStream apiKey = assetManager.open("indicoapitxt.txt");
+            String apiKeyString = CharStreams.toString(new InputStreamReader(apiKey, "UTF-8"));
+            return apiKeyString;
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public void getRedditComment() {
+        try {
+            indico.keywords.predict(spokenString, indicoCallback);
+        } catch (IOException | IndicoException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void getCommentFromKeywords(Set<String> keywords) {
+        String importantWords = StringUtils.join(keywords, " ");
+        commentSearch(importantWords);
+    }
+
+
+    public void commentSearch(String searchQuery) {
+        Uri.Builder builder = new Uri.Builder();
+        builder.scheme("https")
+                .authority("api.pushshift.io")
+                .appendPath("reddit")
+                .appendPath("search")
+                .appendQueryParameter("q", searchQuery)
+                .appendQueryParameter("fields", "body");
+        String Url = builder.build().toString();
+        JsonObjectRequest getRequest = new JsonObjectRequest(Request.Method.GET, Url,
+                new JSONObject(), this, this);
+
+        getRequest.setRetryPolicy(new DefaultRetryPolicy( //changes Volley settings
+                10000, //earlier I was having issues with this api taking more than the 5 seconds it takes Volley to time out
+                DefaultRetryPolicy.DEFAULT_MAX_RETRIES, //now the time is 10 seconds, the api seems faster now
+                DefaultRetryPolicy.DEFAULT_BACKOFF_MULT));
+
+        queue.add(getRequest);
+    }
+
+
+    // TODO: IMPROVE THIS
+    public String pickComment(ArrayList<String> allComments) {
+        for (int i = 0; i < allComments.size(); i++) {
+            if (allComments.get(i).length() < 300) {
+                if (!allComments.get(i).toLowerCase().contains("http")) {
+                    return allComments.get(i);
+                }
+            }
+        }
+        return allComments.get(0);
```

Comment:
instead of returning results when everything you get is too long or has a url in it, why not just give them a toast saying "nothing found"